algebra/src/main/java/org/almostrealism/collect/CollectionFeatures.java

```diff
@@ -115,10 +115,94 @@ static boolean isEnableIndexProjectionDeltaAlt() {
+	 * how data is organized and accessed in collections.
+	 * 
+	 * @param dims the dimensions of the shape (e.g., width, height, depth)
+	 * @return a new {@link TraversalPolicy} representing the specified shape
+	 * 
+	 * @example
+	 * <pre>{@code
+	 * // Create a 1D shape with 5 elements
+	 * TraversalPolicy shape1D = shape(5);
+	 * // Result: shape with dimensions [5], total size = 5
+	 * 
+	 * // Create a 2D shape (matrix) with 3 rows and 4 columns
+	 * TraversalPolicy shape2D = shape(3, 4);
+	 * // Result: shape with dimensions [3, 4], total size = 12
+	 * 
+	 * // Create a 3D shape (tensor) with dimensions 2x3x4
+	 * TraversalPolicy shape3D = shape(2, 3, 4);
+	 * // Result: shape with dimensions [2, 3, 4], total size = 24
+	 * }</pre>
+	 */
 	default TraversalPolicy shape(int... dims) { return new TraversalPolicy(dims); }
+	
+	/**
+	 * Creates a new {@link TraversalPolicy} with the specified dimensions using long values.
+	 * This overload is useful when working with very large dimensions that exceed
+	 * the range of int values.
+	 * 
+	 * @param dims the dimensions of the shape as long values
+	 * @return a new {@link TraversalPolicy} representing the specified shape
+	 * 
+	 * @example
+	 * <pre>{@code
+	 * // Create a large 1D shape
+	 * TraversalPolicy largShape = shape(1000000L);
+	 * // Result: shape with dimensions [1000000], total size = 1000000
+	 * 
+	 * // Create a 2D shape with large dimensions
+	 * TraversalPolicy shape2D = shape(10000L, 20000L);
+	 * // Result: shape with dimensions [10000, 20000], total size = 200000000
+	 * }</pre>
+	 */
 	default TraversalPolicy shape(long... dims) { return new TraversalPolicy(dims); }
+	
+	/**
+	 * Creates a position {@link TraversalPolicy} with the specified dimensions.
+	 * Unlike regular shapes, positions are used to specify coordinates or offsets
+	 * within a larger collection structure.
+	 * 
+	 * @param dims the position coordinates
+	 * @return a new {@link TraversalPolicy} representing the specified position
+	 * 
+	 * @example
+	 * <pre>{@code
+	 * // Create a position at coordinates (2, 3) in a 2D space
+	 * TraversalPolicy pos = position(2, 3);
+	 * // Result: position representing coordinates [2, 3]
+	 * 
+	 * // Create a position at coordinates (1, 2, 3) in a 3D space
+	 * TraversalPolicy pos3D = position(1, 2, 3);
+	 * // Result: position representing coordinates [1, 2, 3]
+	 * }</pre>
+	 */
 	default TraversalPolicy position(int... dims) { return new TraversalPolicy(true, dims); }
 
+	/**
+	 * Extracts the {@link TraversalPolicy} shape from a {@link Supplier}.
+	 * This method is useful for determining the shape of collections at runtime
+	 * by examining the supplier object.
+	 * 
+	 * @param s the supplier to extract shape from
+	 * @return the {@link TraversalPolicy} representing the supplier's shape, or {@link #shape(int...)} if no shape available
+	 * 
+	 * @example
+	 * <pre>{@code
+	 * // Extract shape from a {@link PackedCollection} supplier
```

Comment:
These examples are bad, the examples used should be similar to those in CollectionFeatures::size and focus on CollectionProducer.